### PR TITLE
chore: use module account instead of custom forward address

### PR DESCRIPTION
**Pull request by gjermundgaraba**
## Description

These changes triggered some issues with the transfer module account being blocked by the bank transfers. I solved this by using bankkeeper.SendCoins directly and do a blocked check where used instead.

I have two questions:
1. Would it make sense to put the module addr on the transfer keeper itself to avoid having to look it up all the time?
2. Should the module account + block address check be done in a convenience method instead to avoid all the extra checks everywhere?

closes: #6561

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/keeper/relay.go [hunk 1]
```diff
@@ -209,7 +216,7 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 
 			coin := sdk.NewCoin(token.Denom.IBCDenom(), transferAmount)
 
-			if k.bankKeeper.BlockedAddr(receiver) {
+			if !receiver.Equals(moduleAddr) && k.bankKeeper.BlockedAddr(receiver) {
```
Comment (chatton) [inline modules/apps/transfer/keeper/relay.go:219]: yeah I think something like a `k.IsBlockedAddr(addr)` or `k.IsAllowedToSend(addr)` to handle this check

File: modules/apps/transfer/keeper/forwarding.go [hunk 2]
```diff
@@ -19,8 +19,8 @@ func (k Keeper) forwardPacket(ctx sdk.Context, data types.FungibleTokenPacketDat
 		nextForwardingPath = types.NewForwarding(false, data.Forwarding.Hops[1:]...)
 	}
 
-	// sending from the forward escrow address to the original receiver address.
-	sender := types.GetForwardAddress(packet.DestinationPort, packet.DestinationChannel)
+	// sending from module account (used as a temporary forward escrow) to the original receiver address.
+	sender := k.authKeeper.GetModuleAddress(types.ModuleName)
```
Comment (chatton) [inline modules/apps/transfer/keeper/forwarding.go:23]: looks like `k.authKeeper.GetModuleAddress(types.ModuleName)` is already a map lookup, are you thinking of something like `k.GetModuleAddress()` as a utility to delegate to the authKeeper?

File: modules/apps/transfer/keeper/relay.go [hunk 3]
```diff
@@ -128,8 +129,9 @@ func (k Keeper) sendTransfer(
 			labels = append(labels, telemetry.NewLabel(coretypes.LabelSource, "false"))
 
 			// transfer the coins to the module account and burn them
-			if err := k.bankKeeper.SendCoinsFromAccountToModule(
-				ctx, sender, types.ModuleName, sdk.NewCoins(coin),
+			moduleAccountAddr := k.authKeeper.GetModuleAddress(types.ModuleName) // TODO: Change
+			if err := k.bankKeeper.SendCoins(
+				ctx, sender, moduleAccountAddr, sdk.NewCoins(coin),
```
Comment (chatton) [inline modules/apps/transfer/keeper/relay.go:134]: what's the purpose of this change?

**Review by chatton — COMMENTED** [review]
looking good! I think we should definitely make some sort of helper for the checking of the address, would lean towards a helper fn that delegates to the auth keeper to fetch the address rather than storing a field.

File: modules/apps/transfer/keeper/relay.go (hunk 3 above)
Comment (gjermundgaraba) [reply modules/apps/transfer/keeper/relay.go:134]: Yeah, I don't think this was actually needed, will remove

File: modules/apps/transfer/keeper/relay.go (hunk 1 above)
Comment (gjermundgaraba) [reply modules/apps/transfer/keeper/relay.go]: Done

File: modules/apps/transfer/keeper/forwarding.go (hunk 2 above)
Comment (chatton) [reply modules/apps/transfer/keeper/forwarding.go:23]: happy to have it called directly in the utility fn!

**Review by chatton — APPROVED** [review]
LGTM! Seems a bit cleaner using a module address like this :muscle:

File: modules/apps/transfer/keeper/keeper.go [hunk 4]
```diff
@@ -336,3 +336,14 @@ func (k Keeper) deleteForwardedPacket(ctx sdk.Context, portID, channelID string,
...
 	store.Delete(packetKey)
 }
+
+// IsBlockedAddr checks if the given address is allowed to send or receive tokens.
+// The module account is always allowed to send and receive tokens.
+func (k Keeper) IsBlockedAddr(addr sdk.AccAddress) bool {
+	moduleAddr := k.authKeeper.GetModuleAddress(types.ModuleName)
+	if !addr.Equals(moduleAddr) && k.bankKeeper.BlockedAddr(addr) {
+		return true
+	}
```
[the remark this answers is not in the document]
Comment (gjermundgaraba) [reply modules/apps/transfer/keeper/keeper.go:346]: Ah yes, that is much cleaner! Will do 👍
